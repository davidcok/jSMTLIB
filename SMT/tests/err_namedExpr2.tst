; tests that named expressions are closed
(set-logic AUFLIA)
(declare-fun p () Bool)
(declare-fun q () Bool)
(assert (forall ((r Bool)) (! r :named R))) ; error
(assert (exists ((r Bool)) (! r :named R))) ; error
(assert (forall ((r Bool)) (and r (! q :named R)))) ; OK
(assert (let ((r true)) (! r :named R))) ; error - R already defined
(assert (let ((r true)) (! r :named R2))) ; OK
(assert (forall ((r Bool)) (let ((r true)) (! r :named RR)))) ; OK
(assert (forall ((q Bool)) (let ((r q)) (! r :named RRR)))) ; error - FIXME
