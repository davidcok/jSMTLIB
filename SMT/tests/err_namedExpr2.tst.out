success
success
success
(assert (forall ((r Bool)) (! r :named R))) ; error
                              ^
(error "The expression being named is not closed - this symbol is a variable: r")
(assert (exists ((r Bool)) (! r :named R))) ; error
                              ^
(error "The expression being named is not closed - this symbol is a variable: r")
success
(assert (let ((r true)) (! r :named R))) ; error - R already defined
                                    ^
(error "Symbol R is already defined")
success
success
success
